lib/modules/manager/hermit/artifacts.ts

```diff
@@ -0,0 +1,258 @@
+    contents,
+  };
+}
+
+/**
+ * getAddResult returns the UpdateArtifactsResult for the added files
+ */
+function getAddResult(
+  path: string,
+  contentRes: ReadContentResult
+): UpdateArtifactsResult {
+  return {
+    file: {
+      type: 'addition',
+      path,
+      contents: contentRes.contents,
+      isSymlink: contentRes.isSymlink,
+      isExecutable: contentRes.isExecutable,
+    },
+  };
+}
+
+/**
+ * getDeleteResult returns the UpdateArtifactsResult for deleted files
+ */
+function getDeleteResult(path: string): UpdateArtifactsResult {
+  return {
+    file: {
+      type: 'deletion',
+      path,
+    },
+  };
+}
+
+/**
+ * getUpdateResult will return the update result after `hermit install`
+ * has been performed for all packages
+ */
+async function getUpdateResult(
+  packageFileName: string
+): Promise<UpdateArtifactsResult[]> {
+  const hermitFolder = `${upath.dirname(packageFileName)}/`;
+  const hermitChanges = await getRepoStatus(hermitFolder);
+  logger.debug(
+    { hermitChanges, hermitFolder },
+    `hermit changes after package update`
+  );
+
+  // handle added files
+  const added = await pMap(
+    [...hermitChanges.created, ...hermitChanges.not_added],
+    async (path: string): Promise<UpdateArtifactsResult> => {
+      const contents = await getContent(path);
+
+      return getAddResult(path, contents);
+    },
+    { concurrency: 5 }
+  );
+
+  const deleted = hermitChanges.deleted.map(getDeleteResult);
+
+  const modified = await pMap(
+    hermitChanges.modified,
+    async (path: string): Promise<UpdateArtifactsResult[]> => {
+      const contents = await getContent(path);
+      return [
+        getDeleteResult(path), // delete existing link
+        getAddResult(path, contents), // add a new link
+      ];
+    },
+    { concurrency: 5 }
+  );
+
+  const renamed = await pMap(
+    hermitChanges.renamed,
+    async (renamed): Promise<UpdateArtifactsResult[]> => {
+      const from = renamed.from;
+      const to = renamed.to;
+      const toContents = await getContent(to);
+
+      return [getDeleteResult(from), getAddResult(to, toContents)];
+    },
+    { concurrency: 5 }
+  );
+
+  return [
+    // rename will need to go first, because
+    // it needs to create the new link for the new version
+    // for the modified links to use
+    ...renamed.flat(),
+    ...modified.flat(),
+    ...added,
+    ...deleted,
+  ];
+}
+
+/**
+ * getHermitPackage returns the hermit package for running the hermit install
+ */
+function getHermitPackage(name: string, version: string): string {
+  return `${name}-${version}`;
+}
+
+/**
+ * updateHermitPackage runs hermit install for the given package
+ */
+async function updateHermitPackage(update: UpdateArtifact): Promise<void> {
+  logger.trace({ update }, `hermit.updateHermitPackage()`);
+
+  const toInstall = [];
+  const from = [];
+
+  for (const pkg of update.updatedDeps) {
+    if (!pkg.depName || !pkg.currentVersion || !pkg.newValue) {
+      logger.warn(
```

Comment:
```suggestion
      logger.debug(
```

The thrown error will also cause a error log later